hs_geographic_feature_resource/tests/test_geo_feature.py

```diff
@@ -0,0 +1,580 @@
+
+        value_dict["projection"]="WGS 84 EPSG:4326"
+        value_dict["units"]="Decimal degrees"
+        resource.create_metadata_element(self.resGeoFeature.short_id,'coverage', type='box', value=value_dict)
+
+        # add date of type 'valid'
+        resource.create_metadata_element(self.resGeoFeature.short_id,'date', type='valid', start_date='1/1/2012', end_date='12/31/2012')
+
+        # add a format element
+        format = 'shp'
+        resource.create_metadata_element(self.resGeoFeature.short_id,'format', value=format)
+
+
+        # add a language element
+        #resource.create_metadata_element(self.resGeoFeature.short_id,'language', code='eng')
+
+        # # add 'Publisher' element
+        # original_file_name = 'original.txt'
+        # original_file = open(original_file_name, 'w')
+        # original_file.write("original text")
+        # original_file.close()
+        #
+        # original_file = open(original_file_name, 'r')
+        # # add the file to the resource
+        # hydroshare.add_resource_files(self.resGeoFeature.short_id, original_file)
+
+        # resource.create_metadata_element(self.resGeoFeature.short_id,'publisher', name="HydroShare", url="http://hydroshare.org")
+
+        # add a relation element of uri type
+        resource.create_metadata_element(self.resGeoFeature.short_id,'relation', type='isPartOf',
+                                         value='http://hydroshare.org/resource/001')
+
+        # add another relation element of non-uri type
+        resource.create_metadata_element(self.resGeoFeature.short_id,'relation', type='isDataFor',
+                                         value='This resource is for another resource')
+
+
+        # add a source element of uri type
+        resource.create_metadata_element(self.resGeoFeature.short_id,'source', derived_from='http://hydroshare.org/resource/0002')
+
+        # # add a rights element
+        # resource.create_metadata_element(self.resGeoFeature.short_id,'rights', statement='This is the rights statement for this resource',
+        #                                  url='http://rights.ord/001')
+
+        # add a subject element
+        resource.create_metadata_element(self.resGeoFeature.short_id,'subject', value='sub-1')
+
+        # add another subject element
+        resource.create_metadata_element(self.resGeoFeature.short_id,'subject', value='sub-2')
+        pass
+
+##########################################################################################################3
+
+    def test_geo_feature_res_specific_metadata(self):
+        # originalfileinfo
+        #no originalfileinfo obj
+        self.assertEqual (len(OriginalFileInfo.objects.all()), 0)
+
+        #create originalfileinfo obj without a required para: southlimit
+        self.assertRaises(Exception, lambda: resource.create_metadata_element(self.resGeoFeature.short_id,'OriginalFileInfo', fileType='SHP'))
```

Comment:
the lambda is not necessary. `assertRaises` excepts a function to call and the parameters to pass to it